features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/vitals/FPSVitalListener.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+package com.datadog.android.rum.internal.vitals
+
+import android.os.Build
+import androidx.metrics.performance.FrameData
+import com.datadog.android.core.internal.system.BuildSdkVersionProvider
+import com.datadog.android.rum.internal.domain.FrameMetricsData
+import java.util.concurrent.TimeUnit
+
+internal class FPSVitalListener(
+    private val vitalObserver: VitalObserver,
+    private val buildSdkVersionProvider: BuildSdkVersionProvider = BuildSdkVersionProvider.DEFAULT,
+    private var screenRefreshRate: Double = 60.0
+) : FrameStateListener {
+    private var frameDeadline = EXPECTED_60_FPS_FRAME_DURATION_NS
```

Comment:
should we put `RequiresApi(Build.VERSION_CODES.S)` to this property as well for the clarity?